src/sbtc-operations/commit-reveal-system.md

```diff
@@ -43,14 +43,18 @@ where the first byte is the opcode of the sBTC transaction, the payload is essen
 The reveal transaction is also fairly simple in construction and MUST satisfy the following:
 
 1. It MUST consume an UTXO from a commit transaction as its first input.
-2. The first output MUST be an `OP_RETURN` output with a three byte payload where the first two bytes are the magic bytes (the same ones we promised to add back) that specify the network they are running on - `T2` for mainnet and `X2` for testnet, and the last byte is the opcode `w`.
+2. The first output MUST be an `OP_RETURN` output with a three byte payload where the first two bytes are the magic bytes (the same ones we promised to add back) that specify the network they are running on - `T2` for mainnet and `X2` for testnet, and the last two bytes is an opcode and a script version byte.
     ```
-    0      2  3
-    |------|--|
-    magic  op
+    0      2  3         4
+    |------|--|---------|
+     magic  op  version
     ```
 
-Because the reveal transaction consumes the UTXO from the commit transaction, the data that was embedded in the witness script of the commit transaction is *revealed*. Thus, when the sBTC protocol observes a bitcoin operation with the opcode `w`, it indicates a reveal transaction and the data for the intended operation by the initiator of the commit transaction can be found in the witness of the first input.
+The opcode identifies which type of script is revealed. It is `w` if the script is embedded in segwit witness data, and `r` if the script is in a p2sh redeem script.
+
+The version identifies the SegWit witness version. It is `0` for `p2wsh` scripts and `1` for `p2tr` scripts. If the opcode is `r`, this version byte may be omitted.
```

Comment:
We don't. For `p2sh-p2wsh` this version is already present in the redeem script, so it can be seen as a special `p2sh` transaction with opcode `r` and no extra version byte.